File: lib/src/main/scala/bruto/variability/primitives/StringArgumentVariability.scala
package bruto.variability.primitives

import bruto.variability.{DiscreteSetWalker, ArgumentVariabilityWalker, ArgumentVariability}
import java.lang.reflect.Type

class StringArgumentVariability extends ArgumentVariability {

  private val variants : Array[String] = Array[String](null, "", " ", " a", " a ", "ab", " ab ")


  override def matches(aType: Type) : Boolean = {
    classOf[String] == aType
  }

  override def newWalker(aType: Type) : ArgumentVariabilityWalker = {
    new DiscreteSetWalker(variants)
  }

}

File: lib/src/main/java/bruto/variability/primitives/StringArgumentVariability.java
package bruto.variability.primitives;

import bruto.variability.ArgumentVariability;
import bruto.variability.ArgumentVariabilityWalker;
import bruto.variability.DiscreteSetWalker;

import java.lang.reflect.Type;

public class StringArgumentVariability extends ArgumentVariability {

    private String[] variants = new String[]{null, "", " ", " a", " a ", "ab", " ab "};

    @Override
    public boolean matches(Type aType) {
        return String.class == aType;
    }

    @Override
    public ArgumentVariabilityWalker newWalker(Type aType) {
        return new DiscreteSetWalker(variants);
    }

}
